tests/graph_components/validators/test_finetuning_validator.py

```diff
@@ -162,147 +189,136 @@ def test_validate_after_adding_action_to_domain(
+    # change schema - remove a node
+    schema2 = copy.deepcopy(schema1)
+    schema2.nodes.pop(next(iter(schema2.nodes.keys())))
 
-    # finetuning
+    # finetuning raises - doesn't matter if it's nlu/core/both
     loaded_validate = get_validation_method(
-        finetuning=True, load=True, nlu=nlu, core=core
+        finetuning=True, load=True, nlu=nlu, core=core, graph_schema=schema2,
     )
-
-    # does raise - doesn't matter if it's nlu/core/both
     with pytest.raises(InvalidConfigException):
-        loaded_validate(importer=importer3)
+        loaded_validate(importer=EmptyDataImporter())
 
 
 @pytest.mark.parametrize("nlu, core", [(True, False), (False, True), (True, True)])
-def test_validate_after_changing_core_config(
-    tmp_path: Path,
-    get_validation_method: Callable[..., ValidationMethodType],
-    nlu: bool,
-    core: bool,
+def test_validate_after_adding_node_to_schema(
+    get_validation_method: Callable[..., ValidationMethodType], nlu: bool, core: bool,
 ):
     # training
-    config1 = _get_example_config()
-    importer1 = _create_importer_from_config(config1, tmp_path, "config1.yml")
-    validate = get_validation_method(finetuning=False, load=False, nlu=nlu, core=core)
-    validate(importer=importer1)
+    schema1 = _get_example_schema()
+    schema2 = copy.deepcopy(schema1)
+    schema2.nodes.pop(next(iter(schema2.nodes.keys())))
 
-    # Change Config - remove parts of NLU pipeline
-    config3 = copy.deepcopy(config1)
-    assert len(
-        config3["policies"]
-    ), "Please update the test so that this config has some more policies."
-    config3["policies"] = config3["policies"][:1]  # drop some Policies
-    importer3 = _create_importer_from_config(config3, tmp_path, "config3.yml")
+    validate = get_validation_method(
+        finetuning=False, load=False, nlu=nlu, core=core, graph_schema=schema2
+    )
+    validate(importer=EmptyDataImporter())
 
-    # finetuning
+    # change schema - continue with the schema with one more node than before
+    assert len(schema1.nodes) > len(schema2.nodes)
+
+    # finetuning raises -  doesn't matter if it's nlu/core/both
     loaded_validate = get_validation_method(
-        finetuning=True, load=True, nlu=nlu, core=core
+        finetuning=True, load=True, nlu=nlu, core=core, graph_schema=schema1
     )
-
-    # does raise - doesn't matter if it's nlu/core/both
     with pytest.raises(InvalidConfigException):
-        loaded_validate(importer=importer3)
+        loaded_validate(importer=EmptyDataImporter())
 
 
-class DummyNluDataImporter(NluDataImporter):
-    def __init__(self, messages: List[Message], config: Dict[Text, Any]) -> None:
-        self.training_data = TrainingData(training_examples=messages)
-        self.config = config
-
-    def get_config(self) -> Dict:
-        return self.config
+@pytest.mark.parametrize(
+    "nlu, core, what",
+    [
+        (nlu, core, what)
+        for what in ["uses"]  # , 'needs', 'fn', 'config' ]
```

Comment:
do you want to comment in the other parts as well?

Reply:
yep